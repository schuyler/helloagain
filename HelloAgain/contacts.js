'use strict';

import { Friends } from './models';
let AddressBook = require('react-native-contacts');

export function loadAllContacts() {
    // Load all the contacts from the system addressbook, and within that list,
    // substitute any contacts we already have in our database (and merging any
    // changes from the addressbook). We have to wait until the Contacts model
    // has loaded fully. Returns a Promise that is resolved when everything is
    // loaded.
    return new Promise((resolve, reject) => {
      Friends.hasLoaded.then(() => {
        AddressBook.getAll((err, contacts) => {
          if (err) {
            console.log("Reading addressbook failed:", err);
            reject(err);
            return;
          }
          // Update contacts from the addressbook with whatever we have stored in
          // our database.
          for (var i = 0; i < contacts.length; i++) {
            var storedContact = Friends.load(contacts[i].recordID);
            if (storedContact) {
              Object.assign(storedContact, contacts[i]);
              if (storedContact.isSelected) { // only write back changes to selected contacts
                Friends.save(storedContact);
              }
              contacts[i] = storedContact;
            }
          }
          resolve(contacts);
        });
      });
    });
};
