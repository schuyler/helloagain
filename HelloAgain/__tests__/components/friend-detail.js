'use strict'

import React from 'react';
import renderer from 'react-test-renderer';
import sinon from 'sinon'
import { shallow } from 'enzyme'


import { CONTACT_FIXTURE } from "../fixtures/contacts"
import FriendDetail from "../../components/friend-detail"

it('frienders', () => {
  let fixture = CONTACT_FIXTURE[0]
  expect(renderer.create(
    <FriendDetail item={fixture} />
  )).toMatchSnapshot();
})

it('responds when marked contacted', () => {
  const fixture = CONTACT_FIXTURE[0]
  const onPress = sinon.spy()
  const closeDetail = sinon.spy()
  const wrapper = shallow(
    <FriendDetail item={fixture} onContactedPress={onPress} closeDetail={closeDetail} />
  )
  wrapper.find("Button").simulate("press")
  expect(onPress.calledOnce).toBe(true)
  expect(onPress.args[0][0]).toMatchObject(fixture)
  expect(closeDetail.calledOnce).toBe(true)
})
