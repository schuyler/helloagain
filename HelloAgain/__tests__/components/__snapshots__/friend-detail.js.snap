exports[`test frienders 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "flex": 1,
      "flexDirection": "column",
      "margin": 64,
      "paddingTop": 64,
    }
  }>
  <View
    style={
      Object {
        "flex": 1,
      }
    }>
    <Image
      source={1}
      style={
        Object {
          "height": 128,
          "padding": 32,
          "width": 128,
        }
      }
      uri={undefined} />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "flex": 1,
        "fontSize": 24,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }>
    Nicolaus
     
    Copernicus
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "flex": 1,
        "fontSize": 14,
        "textAlign": "justify",
      }
    }>
    Last contacted: 
    never
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <View
      style={
        Array [
          Object {},
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }>
        Contacted
      </Text>
    </View>
  </View>
</View>
`;
